asserts/gpgkeypairmgr.go

```diff
@@ -298,7 +298,25 @@ func (gkm *GPGKeypairManager) Export(name string) ([]byte, error) {
 	if err != nil {
 		return nil, err
 	}
-	return EncodePublicKey(keyInfo.pubKey)
+	return EncodePublicKey(keyInfo.privKey.PublicKey())
+}
+
+// SignWithoutAuthority assembles an assertion without a set authority with the provided information and signs it with the named key.
+func (gkm *GPGKeypairManager) SignWithoutAuthority(assertType *AssertionType, makeHeaders func(pubKey PublicKey) (map[string]interface{}, error), makeBody func(pubKey PublicKey) ([]byte, error), name string) (Assertion, error) {
+	keyInfo, err := gkm.findByName(name)
+	if err != nil {
+		return nil, err
+	}
+	pubKey := keyInfo.privKey.PublicKey()
+	headers, err := makeHeaders(pubKey)
```

Comment:
I default to being very careful with passing private keys around, but on inspection you're right that GPGKeypairManager.Get already exists, so we might as well have GPGKeypairManager.GetByName too.  I've added that and simplified the rest of it; thanks.